substrate/frame/opf/src/lib.rs

```diff
@@ -0,0 +1,216 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+mod types;
+pub use pallet_scheduler as Schedule;
+pub use types::*;
+
+
+
+#[frame_support::pallet(dev_mode)]
+pub mod pallet {
+	use super::*;
+	use frame_system::WeightInfo;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type RuntimeCall: Parameter
+			+ Dispatchable<RuntimeOrigin = Self::RuntimeOrigin>
+			+ From<Call<Self>>
+			+ IsType<<Self as frame_system::Config>::RuntimeCall>
+			+ From<frame_system::Call<Self>>;
```

Comment:
we no longer need to introduce a new associated type, you can bound those additional constraint direclty on the `system::Config::RuntimeCall`
```suggestion
	pub trait Config: frame_system::Config<RuntimeCall: From<Call<Self>> {
```